CMakeLists.txt

```diff
@@ -38,6 +38,8 @@ option_if_not_defined(CPPDAP_ASAN "Build dap with address sanitizer" OFF)
 option_if_not_defined(CPPDAP_MSAN "Build dap with memory sanitizer" OFF)
 option_if_not_defined(CPPDAP_TSAN "Build dap with thread sanitizer" OFF)
 option_if_not_defined(CPPDAP_INSTALL_VSCODE_EXAMPLES "Build and install dap examples into vscode extensions directory" OFF)
+option_if_not_defined(CPPDAP_USE_EXTERNAL_NLOHMANN_JSON_PACKAGE "Use nlohmann_json with find_package() instead of building internal submodule" OFF)
```

Comment:
Instead of declaring two flags here, could we not have a single `CPPDAP_USE_EXTERNAL_JSON_PACKAGE` option, and when set, require `CPPDAP_JSON_LIBRARY` to be defined?

That way, there's fewer settings that overlap, and less complexity if we were to add another JSON library.

Reply:
Unfortunately, as it stands, trying to do this would ADD more complexity than it reduces. The link library targets differ from the package names, and it is very easy to screw them up, so we would have to hardcode those anyway.

However, I did just reduce complexity by reducing all the json links/includes to a function, that should make it easier if we want to add additional libraries down the line.

Reply:
Ack. Thanks.